crates/fluvio-connector-package/src/secret.rs

```diff
@@ -5,12 +5,18 @@ use std::{
     collections::HashSet,
 };
 
+use anyhow::{Result, anyhow};
 use once_cell::sync::OnceCell;
+use regex::Regex;
 use serde::{Serialize, Deserialize, de::Visitor, Deserializer};
-use anyhow::{Result, anyhow};
-use serde_yaml::Value;
+use serde_yaml::{Value};
 
 static SECRET_STORE: OnceCell<Box<dyn SecretStore>> = OnceCell::new();
+/// Regex for detecting secret references in the config string.
+/// This is used to capture the secret names in the config string.
+/// {{ secrets.SECRET_NAME }}
+/// it returns a capture group with the value SECRET_NAME.
+static SECRETS_RE: OnceCell<Regex> = OnceCell::new();
```

Comment:
not quire sure why regex is being used?

Reply:
We use it to detect which are the secrets that we need to resolve/add to the context

Reply:
but that should be done at context level?

Reply:
secrets resolution should be at SecretStore which is nothing to do with syntax.  SecretStore can be treated like key/value which is orthogonal to Interpolation.

Reply:
yes, I am using this `detect_secrets_from_str` in ContextStore implementation for SecretStore

Reply:
We also use this on cloud to detect which are secrets that we should expose to each connector

Reply:
That is up to SecretStore implementation.  Still not sure why regex is needed.  Interpolation engine should have resolved all syntax.  So it doesn't matter `${{ env.my_value}}` vs `${{ secret.SECRET1}}`.  The context look up should resolve whether we are injecting simple string or Secret.